### PR TITLE
Feature/hide actor action

### PR DESCRIPTION
**Pull request by PhoebeMitchell**
## Summary

Splits the SHOW_ACTOR action into SHOW_ACTOR and HIDE_ACTOR, with an actor name as an optional parameter. This makes the action more conistent with other actions (PLAY/STOP_Song, SHOW/HIDE_ITEM), and also allows for hiding actors when you don't necessarily know the name of the actor to hide, you just want to hide the active actor.

Also adds tests for the SetVisibility method.

## Before/after screenshots and/or animated gif

Kapture 2022-03-24 at 03 33 09

## Testing instructions

Run the tests. Try writing a script with the SHOW/HIDE_ACTOR actions.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:

### CODE REVIEW

File: unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs [hunk 1]
```diff
@@ -63,7 +64,7 @@ public IEnumerator ActorsAreSetToNotTalkingOnDialogueEnd()
         [UnityTest]
         public IEnumerator ActiveSpeakerCanBeSet()
         {
-            var prosecutionAnimator =  GameObject.Find("Prosecution_Actor").GetComponent<Actor>().GetComponent<Animator>();
+            var prosecutionAnimator =  GameObject.Find("Prosecution_Actor").GetComponent<Animator>();
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs:72]: Ooo~, well spotted. The kinda thing I would've expected ReSharper to pick up on. 👍

File: unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs [hunk 2]
```diff
@@ -139,6 +140,30 @@ public IEnumerator ActorNotInTheSceneCanSpeak()
...
+
+        [UnityTest]
+        [TestCase(true, ExpectedResult = true)]
+        [TestCase(false, ExpectedResult = false)]
+        public bool ActorVisibilityCanBeSetForNonActiveActor(bool shouldShow)
+        {
+            return AssertVisibility(shouldShow, "TutorialBoy", GameObject.Find("Prosecution_Actor").GetComponent<Renderer>());
+        }
+
+        private bool AssertVisibility(bool shouldShow, string actorName, Renderer renderer)
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs]: I mean... that's just nifty. 👀👌
Always a big fan of using attributes to generate test data. Makes it super easy to add fuzzing or change test adapters. Not that that's anything Unity is planning on doing any time soon, but looser dependencies tend to be a good thing in my experience. 👍

**Review by ViMaSter — APPROVED** [review]
Better to read, better code coverage... Net plus all around! 👍

File: unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs (hunk 2 above)
Comment (PhoebeMitchell) [reply unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs]: Feel like there should be a way to also combine the renderer used as part of the test data, but TestCase doesn't seem to accept lambdas like `() => GameObject.Find("Prosecution_Actor").GetComponent<Renderer>());` and TestCaseSource would require creating a collection and get's a bit complicated 🤔

Comment (notexplosive) [inline unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs]: nit: `shouldShow` -> `expectedVisibility`

File: unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs [hunk 3]
```diff
@@ -139,6 +140,30 @@ public IEnumerator ActorNotInTheSceneCanSpeak()
...
+        {
+            renderer.enabled = !shouldShow;
+            Assert.IsTrue(renderer.enabled == !shouldShow);
```
Comment (notexplosive) [inline unity-ggjj/Assets/Tests/PlayModeTests/Scripts/ActorControllerTests.cs]: Might be nice if this has a message like "unity sanity check"

**Review by notexplosive — APPROVED** [review]

**Review by lyndsiWilliams — APPROVED** [review]

**Pull request merged**